source/common/http/filter_manager.cc

```diff
@@ -53,6 +53,13 @@ void ActiveStreamFilterBase::commonContinue() {
     return;
   }
 
+  // Set ScopeTrackerScopeState if there's no existing crash context.
+  absl::optional<ScopeTrackerScopeState> state;
+  if (parent_.dispatcher_.emptyTrackedObjectStack()) {
+    const auto& context = restoreContextOnContinue();
+    state.emplace(&context, parent_.dispatcher_);
+  }
```

Comment:
Having continue_context_ as a member variable doesn't feel right.  I wonder if something like:

```
std::vector<ScopeTrackerScopeState> states;
restoreContextOnContinue(states);
```

could be used to construct the appropriate states.  A potential issue is that the objects in states would need to be destructed in reverse order in order to preserve the stack property.  I don't know if vector provides destruction order guarantees; so we may need to wrap this in a class like ScopeTrackerScopeStateStack that provides ways to add more than one scoped object and destroys them in the right order when ScopeTrackerScopeStateStack goes out of scope.

Something like:
```
ScopeTrackerScopeStateStack states;
restoreContextOnContinue(states);
```

Reply:
You're right it doesn't make sense to cache the object since it'd be a waste in the case of filter chains without continuations.

`std::vector` and `absl::inlined_vector` don't guarantee dtor order, so we'll probably have to wrap it.

Reply:
With `OpaqueStateTrackedObject` we avoid the issue of the vector destructor order since only one object is actually added to the stack for potentially many objects.